File: 2022/Python/day1/part1.py
from os import path
basepath = path.dirname(__file__)

maxCalories = 0
with open(path.join(basepath, "input.txt")) as f:
    lines = f.readlines()
    cur = 0
    for line in lines:
        if line == '\n':
            maxCalories = max(cur, maxCalories)
            cur = 0
        else:
            cur += int(line.strip())

print(f"maxCalories: {maxCalories}")

File: 2022/JavaScript/day1/part1.ts
const text = await Deno.readTextFile("./input.txt");

let max = 0;
let current = 0;

const lines = text.split("\n");
for (const line of lines) {
  if (line === "") {
    current = 0;
  } else {
    current += parseInt(line, 10);
    max = Math.max(current, max);
  }
}

console.log(max);
